### PR TITLE
feat: wasm tagrets in multiplatform dsl

### PR DESCRIPTION
**Pull request by danil-pavlov**
- [KT-67308](https://youtrack.jetbrains.com/issue/KT-67308)

### CODE REVIEW

File: docs/topics/multiplatform/multiplatform-dsl-reference.md [hunk 1]
```diff
@@ -65,6 +65,15 @@ The complete list of available targets is the following:
         <td><code>jvm</code></td>
         <td></td>
     </tr>
+    <tr>
+        <td rowspan="2">Kotlin/Wasm</td>
+        <td><code>wasmJs</code></td>
+        <td>Use it for <a href="wasm-js-interop.md">interoperability with JavaScript</a>.</td>
```
Comment (ilgonmic) [inline docs/topics/multiplatform/multiplatform-dsl-reference.md:71]: It is more not about interoperability but about runtimes with JS support
Something like "if you plan to run in JavaScript runtime"

File: docs/topics/multiplatform/multiplatform-dsl-reference.md [hunk 2]
```diff
@@ -122,7 +131,7 @@ In any target block, you can use the following declarations:
 |---------------------|---------------------------------------------------------------------------------------------------------------------------------------------------|
 | `attributes`        | Attributes used for [disambiguating targets](multiplatform-set-up-targets.md#distinguish-several-targets-for-one-platform) for a single platform. |
 | `preset`            | The preset that the target has been created from, if any.                                                                                         |
-| `platformType`      | Designates the Kotlin platform of this target. Available values: `jvm`, `androidJvm`, `js`, `native`, `common`.                                   |
+| `platformType`      | Designates the Kotlin platform of this target. Available values: `jvm`, `androidJvm`, `js`, `native`, `common`, `wasmJs`, `wasmWasi`.             |
```
Comment (ilgonmic) [inline docs/topics/multiplatform/multiplatform-dsl-reference.md:134]: `platformType` are the same for `wasmJs` and `wasmWasi`, just `wasm`, this field is about "which compiler should be used"
https://github.com/JetBrains/kotlin/blob/0e861e386e4b252d4052cd214a9bcefc95249b2d/libraries/tools/kotlin-gradle-plugin-api/src/common/kotlin/org/jetbrains/kotlin/gradle/plugin/KotlinPlatformType.kt#L17

File: docs/topics/multiplatform/multiplatform-dsl-reference.md [hunk 3]
```diff
@@ -147,17 +156,36 @@ kotlin {
...
+| [`nodejs`](#node-js)  | Configuration of the Node.js target. |
 
 Learn more about [configuring Kotlin/JS projects](js-project-setup.md).
 
+A separate `wasmWasi` block describes the configuration of Kotlin/Wasm targets that support the WASI system interface.
+It can contain the following blocks:
+
+| **Name**             | **Description**                                                                                                                | 
+|----------------------|--------------------------------------------------------------------------------------------------------------------------------|
+| [`nodejs`](#node-js) | Configuration of the Node.js target.                                                                                           |
```
Comment (ilgonmic) [inline docs/topics/multiplatform/multiplatform-dsl-reference.md:177]: `nodejs` is now the only option to be used with wasi
`binaries` are available for `js` and `wasmJs` as well, it regulates whether user needs compilation of final binary or not
So `binaries.executable` should be in final module of all targets `js`, `wasmJs` and `wasmWasi`
So I am not sure that we need to describe it specifically in `wasmWasi`

**Review by ilgonmic — CHANGES_REQUESTED** [review]

File: docs/topics/multiplatform/multiplatform-dsl-reference.md (hunk 3 above)
Comment (danil-pavlov) [reply docs/topics/multiplatform/multiplatform-dsl-reference.md]: Thank you! Added  a paragraph about `binaries.executable` for all 3 targets plus a link to a detailed description

**Review by ilgonmic — APPROVED** [review]

**Review by koshachy — APPROVED** [review]
LGTM!

**Pull request merged**